src/mlpack/core/dists/discrete_distribution.hpp

```diff
@@ -153,12 +202,18 @@ class DiscreteDistribution
    *    actually from this distribution.
    */
   void Train(const arma::mat& observations,
-             const arma::vec& probabilities);
+             const arma::mat& probabilities);
 
   //! Return the vector of probabilities.
-  const arma::vec& Probabilities() const { return probabilities; }
+  const arma::vec& Probabilities() const { return probabilities[0]; }
   //! Modify the vector of probabilities.
-  arma::vec& Probabilities() { return probabilities; }
+  arma::vec& Probabilities() { return probabilities[0]; }
+
+  //! Modify the vector Martix of probabilities.
+  const std::vector<arma::vec>& MultipleProbabilities() const { return probabilities; }
+
+  //! Return the vector Martix of probabilities.
+  std::vector<arma::vec>& MultipleProbabilities() { return probabilities; }
```

Comment:
I don't think we should keep `MultipleProbabilities()`; a user can just use the constructor that takes `std::vector<arma::vec>` instead.  So you could adapt the test to be

```
std::vector<arma::vec> pro{{0.1, 0.3, 0.6},
                           {0.3, 0.3, 0.3},
                           {0.25, 0.25, 0.5}};
DiscreteDistribution d(pro);
```